modules/core/depinject.go

```diff
@@ -48,27 +52,53 @@ type ModuleInputs struct {
 type ModuleOutputs struct {
 	depinject.Out
 
-	IbcKeeper *ibckeeper.Keeper
-	Module    appmodule.AppModule
+	Module appmodule.AppModule
+
+	IBCKeeper    *ibckeeper.Keeper
+	ScopedKeeper capabilitykeeper.ScopedKeeper
 }
 
+// ProvideModule defines a depinject provider function to supply the module dependencies and return its outputs.
 func ProvideModule(in ModuleInputs) ModuleOutputs {
 	// default to governance authority if not provided
 	authority := authtypes.NewModuleAddress(govtypes.ModuleName)
 	if in.Config.Authority != "" {
 		authority = authtypes.NewModuleAddressOrBech32Address(in.Config.Authority)
 	}
 
+	scopedKeeper := in.CapabilityKeeper.ScopeToModule(exported.ModuleName)
+
 	keeper := ibckeeper.NewKeeper(
 		in.Cdc,
 		in.Key,
 		in.LegacySubspace,
-		in.ConsensusHost,
+		ibctm.NewConsensusHost(in.StakingKeeper), // NOTE: need to find a way to inject a ConsensusHost into DI container created outside context of app module.
 		in.UpgradeKeeper,
-		in.ScopedKeeper,
+		scopedKeeper,
 		authority.String(),
 	)
 	m := NewAppModule(keeper)
 
-	return ModuleOutputs{IbcKeeper: keeper, Module: m}
+	return ModuleOutputs{Module: m, IBCKeeper: keeper, ScopedKeeper: scopedKeeper}
+}
+
+// InvokeAddAppRoutes defines a depinject Invoker for registering ibc application modules on the core ibc application router.
+func InvokeAddAppRoutes(keeper *ibckeeper.Keeper, appRoutes []porttypes.IBCModuleRoute) {
+	ibcRouter := porttypes.NewRouter()
+
+	for _, route := range appRoutes {
+		ibcRouter.AddRoute(route.Name, route.IBCModule)
+	}
+
+	ibcRouter.Seal()
+}
+
+// InvokeAddClientRoutes defines a depinject Invoker for registering ibc light client modules on the core ibc client router.
+// TODO: Maybe this should align with app router. i.e. create router here, add routes, and set on ibc keeper.
+// For app_v1 this would be the same approach, just create clientRouter in app.go instead of implicit creation inside of ibc.NewKeeper()
+func InvokeAddClientRoutes(keeper *ibckeeper.Keeper, clientRoutes map[string]exported.LightClientModule) {
```

Comment:
Lightclient modules implement `IsOnePerModuleType()` and return the `LightClientModule` in module outputs. Afaik when the DI container has finished providing all modules, then invokers are run. The DI container can collect the `LightClientModules` returned in module outputs into a map, because it is `OnePerModuleType` the key is the module name - e.g. `07-tendermint` 

This is my understanding. TBC when I get this all wired up together with tests!